raiden/tests/integration/test_stress.py

```diff
@@ -0,0 +1,380 @@
+    app = App(
+        config=app.config,
+        chain=app.raiden.chain,
+        query_start_block=0,
+        default_registry=app.raiden.default_registry,
+        default_secret_registry=app.raiden.default_secret_registry,
+        transport=new_transport,
+        raiden_event_handler=RaidenEventHandler(),
+        discovery=app.raiden.discovery,
+    )
+
+    app.start()
+
+    return app
+
+
+def restart_network(raiden_network, retry_timeout):
+    for app in raiden_network:
+        app.stop()
+
+    wait_network = [
+        gevent.spawn(restart_app, app)
+        for app in raiden_network
+    ]
+
+    gevent.wait(wait_network)
+
+    new_network = [
+        greenlet.get()
+        for greenlet in wait_network
+    ]
+
+    # The tests assume the nodes are available to transfer
+    for app0, app1 in combinations(new_network, 2):
+        waiting.wait_for_healthy(
+            app0.raiden,
+            app1.raiden.address,
+            retry_timeout,
+        )
+
+    return new_network
+
+
+def restart_network_and_apiservers(raiden_network, api_servers, port_generator, retry_timeout):
+    """Stop an app and start it back"""
+    for rest_api in api_servers:
+        rest_api.stop()
+
+    new_network = restart_network(raiden_network, retry_timeout)
+    new_servers = start_apiserver_for_network(new_network, port_generator)
+
+    return (new_network, new_servers)
+
+
+def address_from_apiserver(apiserver):
+    return apiserver.rest_api.raiden_api.address
+
+
+def transfer_and_assert(server_from, server_to, token_address, identifier, amount):
+    url = _url_for(
+        server_from,
+        'token_target_paymentresource',
+        token_address=to_checksum_address(token_address),
+        target_address=to_checksum_address(address_from_apiserver(server_to)),
+    )
+    json = {'amount': amount, 'identifier': identifier}
+
+    log.debug('PAYMENT REQUEST', url=url, json=json)
+
+    request = grequests.post(url, json=json)
+    response = request.send().response
+
+    assert (
+        getattr(request, 'exception', None) is None and
+        response is not None and
+        response.status_code == HTTPStatus.OK and
+        response.headers['Content-Type'] == 'application/json'
+    )
+
+
+def sequential_transfers(server_from, server_to, total, token_address, identifier_generator):
+    for _ in range(total):
+        transfer_and_assert(
+            server_from=server_from,
+            server_to=server_to,
+            token_address=token_address,
+            identifier=next(identifier_generator),
+            amount=1,
+        )
+
+
+def stress_send_serial_transfers(rest_apis, token_address, identifier_generator, deposit):
+    """Send `deposit` transfers of value `1` one at the time, without changing
+    the initial capacity.
+    """
+    pairs = list(zip(rest_apis, rest_apis[1:] + [rest_apis[0]]))
+
+    # deplete the channels in one direction
+    for server_from, server_to in pairs:
+        sequential_transfers(
```

Comment:
nitpick: Use kwargs here and below for clearer code comprehension here. Until I followed the code `deposit` looked like the amount that is being transfered.